flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/SqlRewriterUtils.scala

```diff
@@ -130,8 +131,11 @@ object SqlRewriterUtils {
     call.getKind match {
       case SqlKind.SELECT =>
         val sqlSelect = call.asInstanceOf[SqlSelect]
-
-        if (targetPosition.nonEmpty && sqlSelect.getSelectList.size() != targetPosition.size()) {
+        val identifiersSize = sqlSelect.getSelectList.count(s => s.isInstanceOf[SqlIdentifier])
+        if (
+          identifiersSize == 0
```

Comment:
I have a similar question than Jim. What is this line good for? maybe an inline comment makes sense here with an example?

Reply:
continued discussion in https://github.com/apache/flink/pull/25316#discussion_r1829780200